highfidelity/hifi
# Case 20301 - add platform info

https://highfidelity.fogbugz.com/f/cases/20301/Design-and-implement-a-way-to-run-test-ONLY-on-specific-Platform-Profile
# Description
Adds a number of methods to read platform information.  This will be used to enable tests that are platform-specific.
The methods are available from scripts, allowing script writers to check the capabilities of the platform Interface is running on.
# Testing
Run the following tests on both Windows and Mac
1.  Start Interface
2.  Open the JS console
3.  Enter `PlatformInfo.getOperatingSystemType()`
     Verify correct operating system is returned
4.  Enter `PlatformInfo.getCPUBrand()`
     Verify correct CPU brand is returned
5.  Enter `PlatformInfo.getNumLogicalCores()`
     Verify correct number of logical cores is returned
6.  Enter `PlatformInfo.getTotalSystemMemoryMB()`
     Verify that the correct total memory size is returned (in mebibytes)
7.  Enter `PlatformInfo.getGraphicsCardType()`
     Verify that the correct graphics card type is returned.

On Windows with Rift:
1.  Enter `PlatformInfo.hasRiftControllers()`
     Verify that true is returned
2.  Enter `PlatformInfo.hasViveControllers()`
     Verify that false is returned

On Windows with Vive:
1.  Enter `PlatformInfo.hasRiftControllers()`
     Verify that false is returned
2.  Enter `PlatformInfo.hasViveControllers()`
     Verify that true is returned
